### PR TITLE
Add compliant and noncompliant examples of javascript/nosql-i njection@v1.0

### PR DESCRIPTION
**Pull request by Kamlesh-Gour**
*Description of changes:*
* Rules Added: nosql-injection@v1.0
* Added new functions for compliant and non compliant examples
* Added .gitignore

### CODE REVIEW

File: src/detectors/nosql_injection/nosql_injection.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+var app = express()
+function noSqlInjectionCompliant() {
+    app.get('/api/getallusers', function (req, res){
+       var body = req.body
+       // Compliant: should not use external input in `scan` API.
+       var params = {
+       TableName: "dynamodb-example-node",
+       ProjectionExpression: "user_id, username, user_age",
+       }
+       docClient.scan(params, function (err, data) {
```
Comment (ctlai95) [inline src/detectors/nosql_injection/nosql_injection.js:40]: `docClient` initialization is missing from this block

File: src/detectors/nosql_injection/nosql_injection.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+// {/fact}
+
+// {fact rule=nosql-injection@v1.0 defects=0}
+var AWS = require('aws-sdk')
+var express = require("express")
+var app = express()
+function noSqlInjectionCompliant() {
+    app.get('/api/getallusers', function (req, res){
+       var body = req.body
```
Comment (ctlai95) [inline src/detectors/nosql_injection/nosql_injection.js:34]: Looks like this var is not used, can we remove it?

File: src/detectors/nosql_injection/nosql_injection.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+       // Compliant: should not use external input in `scan` API.
+       var params = {
+       TableName: "dynamodb-example-node",
+       ProjectionExpression: "user_id, username, user_age",
+       }
```
Comment (ctlai95) [inline src/detectors/nosql_injection/nosql_injection.js:39]: Please check the indentation/formatting in these lines

File: src/detectors/nosql_injection/nosql_injection.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        // Noncompliant: external user input can be vulnerable to injection attacks.
+        dobClient.query(params, function(err, data) {
+         if (err) {
+            console.log("Error", err)
+         } else {
+            data.Items.forEach(function(element, index, array) {
+            console.log(element.Title.S + " (" + element.Subtitle.S + ")")
+            })
+         }
+        })
```
Comment (ctlai95) [inline src/detectors/nosql_injection/nosql_injection.js:23]: Please check the indentation/formatting in these lines

File: src/detectors/nosql_injection/nosql_injection.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+       TableName: "dynamodb-example-node",
+       ProjectionExpression: "user_id, username, user_age",
+       }
+       docClient.scan(params, function (err, data) {
+       if (err) {
+         console.log(err)
+       } else {
+         res.status(200).json({ "status": 1, "message": "user exists", "data": data.Items })
+       }
+       })
```
Comment (ctlai95) [inline src/detectors/nosql_injection/nosql_injection.js:46]: Please check the indentation/formatting in these lines

**Review by ctlai95 — CHANGES_REQUESTED** [review]

File: src/detectors/nosql_injection/nosql_injection.js (hunk 4 above)
Comment (Kamlesh-Gour) [reply src/detectors/nosql_injection/nosql_injection.js]: Corrected indentation

File: src/detectors/nosql_injection/nosql_injection.js (hunk 3 above)
Comment (Kamlesh-Gour) [reply src/detectors/nosql_injection/nosql_injection.js]: Corrected indentation

File: src/detectors/nosql_injection/nosql_injection.js (hunk 5 above)
Comment (Kamlesh-Gour) [reply src/detectors/nosql_injection/nosql_injection.js]: Corrected indentation

File: src/detectors/nosql_injection/nosql_injection.js (hunk 2 above)
Comment (Kamlesh-Gour) [reply src/detectors/nosql_injection/nosql_injection.js]: Removed unused `var`

File: src/detectors/nosql_injection/nosql_injection.js (hunk 1 above)
Comment (Kamlesh-Gour) [reply src/detectors/nosql_injection/nosql_injection.js:40]: Added `docClient` initialization

File: src/detectors/nosql_injection/nosql_injection.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// {fact rule=nosql-injection@v1.0 defects=1}
+var AWS = require('aws-sdk')
+var express = require("express")
+var app = express()
+function noSqlInjectionNoncompliant() {
+    app.get(path, function(req,res) {
```
Comment (ctlai95) [inline src/detectors/nosql_injection/nosql_injection.js:11]: Should it be `/api/getallusers` to keep it consistent?

File: src/detectors/nosql_injection/nosql_injection.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        var dobClient = new AWS.DynamoDB({apiVersion: '2012-08-10'})
+        var params= req.body.params
+        // Noncompliant: external user input can be vulnerable to injection attacks.
+        dobClient.query(params, function(err, data) {
```
Comment (ctlai95) [inline src/detectors/nosql_injection/nosql_injection.js:15]: Can we use `docClient.scan` to keep it consistent?

File: src/detectors/nosql_injection/nosql_injection.js (hunk 6 above)
Comment (Kamlesh-Gour) [reply src/detectors/nosql_injection/nosql_injection.js]: Updated

File: src/detectors/nosql_injection/nosql_injection.js (hunk 7 above)
Comment (Kamlesh-Gour) [reply src/detectors/nosql_injection/nosql_injection.js]: Updated

**Review by mj632 — APPROVED** [review]

**Pull request merged**